src/DataGrid.tsx

```diff
@@ -799,13 +799,18 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getDragHandle(rowIdx: number) {
-    if (selectedPosition.rowIdx !== rowIdx || selectedPosition.mode === 'EDIT' || onFill == null) {
+  function renderDragHandle() {
+    if (
```

Comment:
Do we need to check whether the selected cell is not in a summary row?

Reply:
`isCellWithinViewportBounds` would return false for header and summary rows. There is another function `isCellWithinSelectionBounds` that can be used to check header and summary rows as well